### PR TITLE
SC-3258, feature: Add sync button+dialog for both bucket+object respe…

### PR DESCRIPTION
**Pull request by wilwong89**
# Description

Add sync button to bucket list page for each bucket.
That button will trigger a dialog box to start bucket sync.

Add sync button to object list page for each object.
That button will trigger a dialog box to start object sync.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

Needs Tyler to check if UI is to spec. Screenshots included in https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3258
Needs someone to double check sync calls and syncing feature works as expected.

### CODE REVIEW

**Review by jujaga — COMMENTED** [review]
Visually seems correct. A few comments on code structure however:
- We should not be in-lining the syncBucket dialog and action behavior as a part of the BucketList component; this should be a standalone component at minimum similar to your new SyncObjectButton.
- Followup from above: we should unify both the SyncObjectButton and the new SyncBucket equivalent to both point to one component called "SyncButton" which resides under root `/components` folder. This component should take in two optional parameters; either an objectId, or a bucketId; depending on which was passed in, it will know to call either syncObject or syncBucket store methods appropriately.

File: frontend/src/components/common/SyncButton.vue [hunk 1]
```diff
@@ -1,38 +1,48 @@
...
+const bucketStore = useBucketStore();
+const toast = useToast();
 
 // Dialog
 const displaySyncDialog = ref(false);
 
 // Actions
-const syncObject = () => {
-  objectStore.syncObject(props.id);
+const syncFunction = () => {
```
Comment (jujaga) [inline frontend/src/components/common/SyncButton.vue:30]: Might be better to just call this onSubmit instead

File: frontend/src/components/common/SyncButton.vue [hunk 2]
```diff
@@ -1,38 +1,48 @@
...
 
-import type { Ref } from 'vue';
-import type { COMSObject } from '@/types';
 
 // Props
 type Props = {
+  type: string;
   id: string;
+  name: string;
 };
```
Comment (jujaga) [inline frontend/src/components/common/SyncButton.vue:14]: Is there a reason why we didn't do props like this instead?
```ts
type Props = {
  bucketId?: string,
  objectId?: string
};
```
Forcing a name dependency upstream as a required prop doesn't appear to be a great idea; the component should be the one being able to figure out that information itself. Also if we want to be doing an explicit type string instead, this should be defined as an enum type, not as two constant strings.

File: frontend/src/components/object/ObjectTable.vue [hunk 3]
```diff
@@ -9,7 +9,8 @@ import {
   ObjectFilters,
   ObjectPermission,
 } from '@/components/object';
-import { ShareObjectButton, SyncObjectButton } from '@/components/object/share';
+import { ShareObjectButton } from '@/components/object/share';
+import { SyncButton } from '@/components/common';
```
Comment (jujaga) [inline frontend/src/components/object/ObjectTable.vue:13]: Switch the import ordering - we sort based on the from string, not the component name.

File: frontend/src/components/common/SyncButton.vue [hunk 4]
```diff
@@ -0,0 +1,115 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import { Button, Dialog, useToast } from '@/lib/primevue';
+import { useObjectStore, useBucketStore } from '@/store';
+
+
```
Comment (jujaga) [inline frontend/src/components/common/SyncButton.vue:8]: Extra unnecessary whitespace line

File: frontend/src/store/bucketStore.ts [hunk 5]
```diff
@@ -118,7 +133,8 @@ export const useBucketStore = defineStore('bucket', () => {
     deleteBucket,
     fetchBuckets,
     findBucketById,
-    updateBucket
+    updateBucket,
+    syncBucket
```
Comment (jujaga) [inline frontend/src/store/bucketStore.ts:137]: This should be in alphabetical order.

File: frontend/src/store/objectStore.ts [hunk 6]
```diff
@@ -266,7 +280,8 @@ export const useObjectStore = defineStore('object', () => {
     headObject,
     setSelectedObjects,
     togglePublic,
-    updateObject
+    updateObject,
+    syncObject
```
Comment (jujaga) [inline frontend/src/store/objectStore.ts:284]: Alphabetical order

File: frontend/src/components/common/SyncButton.vue [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    <font-awesome-icon icon="fa-solid fa-sync" />
+  </Button>
+</template>
+
+<style scoped lang="scss">
+h2 {
+  font-weight: bold;
+}
+ul {
+  padding-left: 22px;
```
Comment (jujaga) [inline frontend/src/components/common/SyncButton.vue:113]: Unless this was explicitly defined, it may be better to have this be defined in either rem or em units instead of explicit pixels to maintain relativity calculations.

**Review by jujaga — APPROVED** [review]

**Review by TimCsaky — APPROVED** [review]
good work!

**Review by norrisng-bc — APPROVED** [review]
